cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1597,21 +1593,22 @@ makeTransactionBody =
 
 
 pattern TxBody :: TxBodyContent ViewTx era -> TxBody era
-pattern TxBody txbodycontent <- (getTxBodyContent -> txbodycontent)
+pattern TxBody txbodycontent <- (getTxBodyContent TxScriptValidityNone -> txbodycontent)
 {-# COMPLETE TxBody #-}
 
-getTxBodyContent :: TxBody era -> TxBodyContent ViewTx era
-getTxBodyContent (ByronTxBody body) = getByronTxBodyContent body
-getTxBodyContent (ShelleyTxBody era body _scripts _redeemers mAux _scriptValidity) =
-    fromLedgerTxBody era body mAux
+getTxBodyContent :: TxScriptValidity era -> TxBody era -> TxBodyContent ViewTx era
+getTxBodyContent _ (ByronTxBody body) = getByronTxBodyContent body
+getTxBodyContent scriptValidity (ShelleyTxBody era body _scripts _redeemers mAux _scriptValidity) =
+    fromLedgerTxBody era scriptValidity body mAux
```

Comment:
No we would have to add the validity flag into the `TxBody` type, probably as an extra field in the `ShelleyTxBody` constructor. We have to get the value from the underlying ledger tx type, and not invent it. For one thing we need to do that to have the TxBody round trip test pass, but more importantly we need it to get the right value for the tx from a block.

Reply:
See also `getAlonzoTxBody` where we extract the parts of the `Alonzo.ValidatedTx`.

Reply:
I remove the argument of `getTxBodyContent` and am taking it from `TxBody`.